File: LeetCode/problems/p0026/java/src/p026/java/P026.java
//[COMPLETED]

package p026;

import mathtools.MF;

public class P026 {

    public static void main(String[] args) {
        Solution s = new Solution();
        int[] arr = new int[] {1,1,2};
        System.out.println(s.removeDuplicates(arr));
        MF.printList(arr);
    }

    
private static class Solution {
    public int removeDuplicates(int[] nums) {
        int place = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == nums[place])
                continue;
            place++;
            nums[place] = nums[i];
        }
        //don't need this
        for (int i = place + 1; i < nums.length; i++)
            nums[i] = 0;
        return place + 1;
    }
}
}

File: leet-code/src/problems/p0026/python/p026.py
#[COMPLETED]

class Solution:
    def removeDuplicates(self, nums) -> int:
        if len(nums) == 0:
            return 0
        num_unique = 0
        i = 0
        while True:
            save_i = i
            i += 1
            while i < len(nums) and nums[i] == nums[save_i]:
                i += 1
            nums[num_unique] = nums[save_i]
            num_unique += 1
            if i >= len(nums):
                break
        return num_unique

sol = Solution()
arr = [1,1,2,2,3,3,3,3,3,3]
print(sol.removeDuplicates(arr))
print(arr)